mlflow/store/model_registry/sqlalchemy_store.py

```diff
@@ -446,19 +446,28 @@ def transition_model_version_stage(self, name, version, stage,
 
         :return: A single :py:class:`mlflow.entities.model_registry.ModelVersion` object.
         """
-        if archive_existing_versions:
-            raise MlflowException("'archive_existing_versions' flag is not supported in "
-                                  "SqlAlchemyStore. Set it to 'False'")
+        is_active_stage = get_canonical_stage(stage) in DEFAULT_STAGES_FOR_GET_LATEST_VERSIONS
+        if archive_existing_versions and not is_active_stage:
+            msg_tpl = ("Model version transition cannot archive existing model versions "
+                       "because '{}' is not an Active stage. Valid stages are {}")
+            raise MlflowException(msg_tpl.format(stage, DEFAULT_STAGES_FOR_GET_LATEST_VERSIONS))
+
         with self.ManagedSessionMaker() as session:
+            last_updated_time = now()
+            in_same_stage = SqlModelVersion.current_stage == stage
+            model_versions = session.query(SqlModelVersion).filter(in_same_stage).all()
```

Comment:
Is this scoped to a single registered model?

Reply:
mm might want to add a test that other registered model versions are not affected

Reply:
Thanks for pointing this out. I fixed the conditin as follows:

```python
conditions = [
    SqlModelVersion.name == name,
    SqlModelVersion.version != version,
    SqlModelVersion.current_stage == stage,
]
model_versions = session.query(SqlModelVersion).filter(in_same_stage).all()
```